### PR TITLE
State root + precompile-ecc

### PR DESCRIPTION
**Pull request by cassc**
This PR contains changes from the other two PRs [state-root](https://github.com/sbip-sg/CuEVM/pull/23/files) and [precompile-ecc](https://github.com/sbip-sg/CuEVM/pull/22)

### CODE REVIEW

**Comment by minhhn2910** [comment]
We need to post the stats on passed test first (directly as a section in the Readme file). Then disable printing before merging. We can debug the minority corner cases and GPU in another PR.

**Comment by cassc** [comment]
@minhhn2910 tests results added in readme:
https://github.com/sbip-sg/CuEVM/tree/state-root-ecc?tab=readme-ov-file#test-results

File: src/state.cuh [hunk 1]
```diff
@@ -2152,8 +2188,11 @@ public:
                 );
                 _content->touch = tmp_touch;
                 _content->touch[account_idx] = 0;
-                delete dup_account;
             )
+
+            delete dup_account;
+            dup_account = nullptr;
```
Comment (sdcioc) [inline src/state.cuh:2194]: Shared memory must be free just by one thread if allocated by one thread. I will fix it.

File: src/state.cuh [hunk 2]
```diff
@@ -2406,6 +2451,8 @@ public:
     )
     {
         bn_t gas_cost, set_gas_refund;
+
+        // todo missing check for eip-1706?
```
Comment (sdcioc) [inline src/state.cuh:2455]: I check for it in the operation. eip-2200 is saying to verify the gas left before computing the gas cost/refund for sstore operation, which a little different than the initial eip-1706.
